tests/testing-utils/src/chain_clients.rs

```diff
@@ -422,59 +493,112 @@ pub async fn create_total_delegated_stake_operator_set(
     let service_manager_address = get_service_manager_address(http_endpoint.to_string()).await;
     let service_manager =
         MockAvsServiceManager::new(service_manager_address, default_signer.clone());
-
-    service_manager
-        .setAppointee(
+    let permission_controller_address =
+        get_permission_controller_address(http_endpoint.to_string()).await;
+    let contract_permission_controller =
+        PermissionController::new(permission_controller_address, get_provider(http_endpoint));
+    if !contract_permission_controller
+        .canCall(
+            service_manager_address,
             default_signer.default_signer_address(),
             allocation_manager_addr,
             alloy::primitives::FixedBytes(AllocationManager::setAVSRegistrarCall::SELECTOR),
         )
-        .send()
+        .call()
         .await
         .unwrap()
-        .get_receipt()
-        .await
-        .unwrap();
+        ._0
+    {
+        service_manager
+            .setAppointee(
+                default_signer.default_signer_address(),
+                allocation_manager_addr,
+                alloy::primitives::FixedBytes(AllocationManager::setAVSRegistrarCall::SELECTOR),
+            )
+            .send()
+            .await
+            .unwrap()
+            .get_receipt()
+            .await
+            .unwrap();
+    }
 
     let registry_coordinator_addr =
         get_registry_coordinator_address(http_endpoint.to_string()).await;
 
-    allocation_manager
-        .setAVSRegistrar(avs_address, registry_coordinator_addr)
-        .send()
+    if allocation_manager
+        .getAVSRegistrar(avs_address)
+        .call()
         .await
         .unwrap()
-        .get_receipt()
-        .await
-        .unwrap();
+        ._0
+        .eq(&avs_address)
```

Comment:
We can use `==` instead of `eq`

Reply:
https://github.com/Layr-Labs/eigensdk-rs/pull/420/commits/13206aa836a9fc5469464ce3514d078ab8bfde6b